modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/monitor/Monitor.java

```diff
@@ -11,67 +11,205 @@
+            throw new EntityNotFoundException("Resource not found.");
+        }
+
+        if (!DeployResourceKind.VM.equals(resourceEntity.getKind())) {
+            String errorMsg =
+                    String.format("Resource kind %s not support.", resourceEntity.getKind());
+            log.error(errorMsg);
+            throw new RuntimeException(errorMsg);
+        }
+        DeployResource deployResource = new DeployResource();
+        BeanUtils.copyProperties(resourceEntity, deployResource);
+        DeployServiceEntity serviceEntity = findDeployServiceEntity(resourceEntity.getId());
+        CredentialDefinition credential =
+                getCredential(serviceEntity.getCsp(), serviceEntity.getUserName());
+        OrchestratorPlugin orchestratorPlugin =
+                orchestratorService.getOrchestratorPlugin(serviceEntity.getCsp());
+        ResourceMetricRequest resourceMetricRequest =
+                getResourceMetricRequest(deployResource, credential, monitorType, from,
+                        to, period);
+
+        return orchestratorPlugin.getMetricsForResource(resourceMetricRequest);
+    }
+
+
+    private DeployServiceEntity findDeployServiceEntity(UUID id) {
+        DeployServiceEntity serviceEntity =
+                deployServiceStorage.findDeployServiceById(id);
+        if (Objects.isNull(serviceEntity)) {
+            throw new EntityNotFoundException("Service not found.");
+        }
+        return serviceEntity;
+    }
+
+    private CredentialDefinition getCredential(Csp csp, String userName) {
+        CredentialDefinition credential = credentialCenter.getCredential(
+                csp, userName);
+        if (Objects.isNull(credential)) {
+            throw new EntityNotFoundException("Credential not found.");
+        }
+        return credential;
+    }
+
+
+    private ResourceMetricRequest getResourceMetricRequest(DeployResource deployResource,
+                                                           CredentialDefinition credential,
+                                                           MonitorResourceType monitorType,
+                                                           Long from, Long to, Integer period) {
+        if (Objects.isNull(from)) {
+            from = System.currentTimeMillis() - 60 * 1000;
+        }
+        if (Objects.isNull(to)) {
+            to = System.currentTimeMillis();
+        }
+        return new ResourceMetricRequest(deployResource, credential, monitorType, from, to, period);
+    }
+
+    private ServiceMetricRequest getServiceMetricRequest(List<DeployResource> deployResources,
+                                                         CredentialDefinition credential,
+                                                         MonitorResourceType monitorType,
+                                                         Long from, Long to, Integer period) {
+        if (Objects.isNull(from)) {
+            from = System.currentTimeMillis() - 60 * 1000;
```

Comment:
we must document this default value in the API also.

Reply:
I will add it.